### ISSUE TITLE
Assertion triggered, representative zero

### ISSUE THREAD

**Issue by paulmelis** [nanocurrency/raiblocks#751]
**Description of bug:**

On a fresh sync (i.e. empty ~/RaiBlocks dir) I get an assertion failure in a debug build:

```
nano_wallet: /home/melis/c/raiblocks-git/rai/blockstore.cpp:547: void rai::block_store::representation_add(MDB_txn*, const block_hash&, const uint128_t&): Assertion `!source_rep.is_zero ()' failed.

Thread 13 "nano_wallet" received signal SIGABRT, Aborted.
[Switching to Thread 0x7effbb7fe700 (LWP 15100)]
0x00007ffff5bc4860 in raise () from /usr/lib/libc.so.6
(gdb) bt
#0  0x00007ffff5bc4860 in raise () at /usr/lib/libc.so.6
#1  0x00007ffff5bc5ec9 in abort () at /usr/lib/libc.so.6
#2  0x00007ffff5bbd0bc in __assert_fail_base () at /usr/lib/libc.so.6
#3  0x00007ffff5bbd133 in  () at /usr/lib/libc.so.6
#4  0x0000555555d620d2 in rai::block_store::representation_add(MDB_txn*, rai::uint256_union const&, boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<128u, 128u, (boost::multiprecision::cpp_integer_type)0, (boost::multiprecision::cpp_int_check_type)0, void>, (boost::multiprecision::expression_template_option)0> const&) (this=0x5555569e0668, transaction_a=0x555556ceb890, source_a=..., amount_a=...)
    at /home/melis/c/raiblocks-git/rai/blockstore.cpp:547
#5  0x0000555555d731e3 in (anonymous namespace)::ledger_processor::open_block(rai::open_block const&) (this=0x7effbb7fd5a0, block_a=...) at /home/melis/c/raiblocks-git/rai/ledger.cpp:467
#6  0x0000555555dec0da in rai::open_block::visit(rai::block_visitor&) const (this=0x7eff96ff6a40, visitor_a=...) at /home/melis/c/raiblocks-git/rai/lib/blocks.cpp:540
#7  0x0000555555d73cf6 in rai::ledger::process(MDB_txn*, rai::block const&) (this=0x5555569e07e0, transaction_a=0x555556ceb890, block_a=...) at /home/melis/c/raiblocks-git/rai/ledger.cpp:576
#8  0x0000555555b777c6 in rai::block_processor::process_receive_one(MDB_txn*, std::shared_ptr<rai::block>) (this=0x5555569e1e10, transaction_a=0x555556ceb890, block_a=std::shared_ptr (count 2, weak 0) 0x7eff96ff6a40)
    at /home/melis/c/raiblocks-git/rai/node/node.cpp:1292
#9  0x0000555555b77198 in rai::block_processor::process_receive_many(std::deque<rai::block_processor_item, std::allocator<rai::block_processor_item> >&) (this=0x5555569e1e10, blocks_processing=std::deque with 60 elements = {...})
    at /home/melis/c/raiblocks-git/rai/node/node.cpp:1250
#10 0x0000555555b76b92 in rai::block_processor::process_blocks() (this=0x5555569e1e10) at /home/melis/c/raiblocks-git/rai/node/node.cpp:1205
#11 0x0000555555b79acf in rai::node::<lambda()>::operator()(void) const (__closure=0x5555568e8228) at /home/melis/c/raiblocks-git/rai/node/node.cpp:1445
#12 0x0000555555b946ed in std::__invoke_impl<void, rai::node::node(rai::node_init&, boost::asio::io_service&, const boost::filesystem::path&, rai::alarm&, const rai::node_config&, rai::work_pool&)::<lambda()> >(std::__invoke_other, rai::node::<lambda()> &&) (__f=...) at /usr/include/c++/7.3.0/bits/invoke.h:60
#13 0x0000555555b91c00 in std::__invoke<rai::node::node(rai::node_init&, boost::asio::io_service&, const boost::filesystem::path&, rai::alarm&, const rai::node_config&, rai::work_pool&)::<lambda()> >(rai::node::<lambda()> &&) (__fn=...)
    at /usr/include/c++/7.3.0/bits/invoke.h:95
#14 0x0000555555bc7b4e in std::thread::_Invoker<std::tuple<rai::node::node(rai::node_init&, boost::asio::io_service&, const boost::filesystem::path&, rai::alarm&, const rai::node_config&, rai::work_pool&)::<lambda()> > >::_M_invoke<0>(std::_Index_tuple<0>) (this=0x5555568e8228) at /usr/include/c++/7.3.0/thread:234
#15 0x0000555555bc79aa in std::thread::_Invoker<std::tuple<rai::node::node(rai::node_init&, boost::asio::io_service&, const boost::filesystem::path&, rai::alarm&, const rai::node_config&, rai::work_pool&)::<lambda()> > >::operator()(void) (this=0x5555568e8228) at /usr/include/c++/7.3.0/thread:243
#16 0x0000555555bc7892 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<rai::node::node(rai::node_init&, boost::asio::io_service&, const boost::filesystem::path&, rai::alarm&, const rai::node_config&, rai::work_pool&)::<lambda()> > > >::_M_run(void) (this=0x5555568e8220) at /usr/include/c++/7.3.0/thread:186
#17 0x000055555606cc3f in execute_native_thread_routine ()
#18 0x00007ffff736708c in start_thread () at /usr/lib/libpthread.so.0
#19 0x00007ffff5c85e7f in clone () at /usr/lib/libc.so.6
(gdb) up
#1  0x00007ffff5bc5ec9 in abort () from /usr/lib/libc.so.6
(gdb) 
#2  0x00007ffff5bbd0bc in __assert_fail_base () from /usr/lib/libc.so.6
(gdb) 
#3  0x00007ffff5bbd133 in __assert_fail () from /usr/lib/libc.so.6
(gdb) 
#4  0x0000555555d620d2 in rai::block_store::representation_add (this=0x5555569e0668, transaction_a=0x555556ceb890, source_a=..., amount_a=...) at /home/melis/c/raiblocks-git/rai/blockstore.cpp:547
547		assert (!source_rep.is_zero ());
(gdb) p source_a
$5 = (const rai::block_hash &) @0x7effbb7fd380: {bytes = {_M_elems = "\353\356ZNĶ\315y\\\253gC2\265\243\025@3\365~\347.\t\244\177eZ\217M\377", <incomplete sequence \305>}, chars = {
    _M_elems = "\353\356ZNĶ\315y\\\253gC2\265\243\025@3\365~\347.\t\244\177eZ\217M\377", <incomplete sequence \305>}, dwords = {_M_elems = {1314582251, 2043524804, 1130867548, 363050290, 2129998656, 2752065255, 2405066111, 
      3311402829}}, qwords = {_M_elems = {8776872203059392235, 1559289123484183388, 11820030268812899136, 14222366856841946495}}, owords = {_M_elems = {{bytes = {_M_elems = "\353\356ZNĶ\315y\\\253gC2\265\243\025"}, chars = {
          _M_elems = "\353\356ZNĶ\315y\\\253gC2\265\243\025"}, dwords = {_M_elems = {1314582251, 2043524804, 1130867548, 363050290}}, qwords = {_M_elems = {8776872203059392235, 1559289123484183388}}}, {bytes = {
          _M_elems = "@3\365~\347.\t\244\177eZ\217M\377", <incomplete sequence \305>}, chars = {_M_elems = "@3\365~\347.\t\244\177eZ\217M\377", <incomplete sequence \305>}, dwords = {_M_elems = {2129998656, 2752065255, 2405066111, 
            3311402829}}, qwords = {_M_elems = {11820030268812899136, 14222366856841946495}}}}}}
(gdb) p source_a.c_str()
Couldn't find method rai::block_hash::c_str
(gdb) p source_a.to_string()
$6 = "EBEE5A4EC4B6CD795CAB674332B5A3154033F57EE72E09A47F655A8F4DFF5FC5"
(gdb) p amount_a
$7 = (const rai::uint128_t &) @0x7effbb7fd370: {m_backend = {<boost::multiprecision::backends::cpp_int_base<128, 128, (boost::multiprecision::cpp_integer_type)0, (boost::multiprecision::cpp_int_check_type)0, void, true>> = {
      static limb_bits = 128, static limb_mask = 340282366920938463463374607431768211455, m_data = 2080000000000000000000000000000}, <No data fields>}}

```

**Steps to reproduce the issue:**
1. rm -rf ~/RaiBlocks
2. $ nano_wallet

**Additional information you deem important (e.g. issue happens only occasionally):**

I actually got the assertion on the very first bootstrap but failed to get a core dump, restarted nano_wallet under gdb, got it again.

**Environment**:

Linux juggle 4.15.8-1-ARCH #1 SMP PREEMPT Sat Mar 10 00:00:33 UTC 2018 x86_64 GNU/Linux

master a22ceb5f7bd07cd00324d36a753f92847f6b54cf

**logs**

Here's the relevant log file, but I see nothing strange on a first glance: 
[log_2018-03-24_18-06-07.2.log](https://github.com/nanocurrency/raiblocks/files/1844661/log_2018-03-24_18-06-07.2.log)

**Comment by paulmelis** [comment]
This is the open block: https://nano.org/en/explore/block/EBEE5A4EC4B6CD795CAB674332B5A3154033F57EE72E09A47F655A8F4DFF5FC5

Here's the source block: https://nano.org/en/explore/block/EBEE5A4EC4B6CD795CAB674332B5A3154033F57EE72E09A47F655A8F4DFF5FC5?h=undefined

Its representative is set to the burn account, that's probaby it?

**Comment by cryptocode** [comment]
Yeah, and it'll be ignored in release builds.

**Comment by PlasmaPower** [comment]
Fixed with #753.
